lib/diagnostic.js

```diff
@@ -5,17 +5,24 @@
 // the minimum value in an array
 // this callback will be passed by the tests to the reduce method, e.g.
 // `let minimum = array.reduce(min);`
-const min = function (/* previous, current */) {
+const min = function (previous, current) {
+  for (let i = 0; i < previous.length; i++) {
+    if (current > previous[i]) {
+      current = previous[i];
+    }
+  }
 
+  return current;
 };
 
 // Question 2
 // Write a callback for reduce that produces
 // the sum of the values in the array
 // this callback will be passed by the tests to the reduce method, e.g.
 // `let total = array.reduce(sum);`
-const sum = function (/* previous, current */) {
-
+const sum = function (previous, current) {
+  let add = previous.reduce(previous + current);
```

Comment:
```previous``` can be anything and if it's not an array or something that implements ```reduce```, then it'll throw an error.